test/mongo/container.spec.ts

```diff
@@ -0,0 +1,105 @@
+import "expect-puppeteer";
+import crypto from "crypto";
+import { login } from "../utils/shared";
+
+jest.setTimeout(300000);
+
+const LOADING_STATE_DELAY = 2500;
+const RENDER_DELAY = 1000;
+
+describe("Collection Add and Delete Mongo spec", () => {
+  it("creates and deletes a collection", async () => {
+    try {
+      const dbId = `TestDatabase${crypto.randomBytes(8).toString("hex")}`;
+      const collectionId = `TestCollection${crypto.randomBytes(8).toString("hex")}`;
+      const sharedKey = `SharedKey${crypto.randomBytes(8).toString("hex")}`;
+      const frame = await login(process.env.MONGO_CONNECTION_STRING);
+
+      // create new collection
+      await frame.waitFor('button[data-test="New Collection"]', { visible: true });
+      await frame.waitForSelector('div[class="splashScreen"] > div[class="title"]', { visible: true });
+      await frame.click('button[data-test="New Collection"]');
+
+      // check new database
+      await frame.waitFor('input[data-test="addCollection-createNewDatabase"]');
+      await frame.click('input[data-test="addCollection-createNewDatabase"]');
+
+      // check shared throughput
+      await frame.waitFor('input[data-test="addCollectionPane-databaseSharedThroughput"]');
+      await frame.click('input[data-test="addCollectionPane-databaseSharedThroughput"]');
+
+      // type database id
+      await frame.waitFor('input[data-test="addCollection-newDatabaseId"]');
+      await frame.type('input[data-test="addCollection-newDatabaseId"]', dbId);
+
+      // type collection id
+      await frame.waitFor('input[data-test="addCollection-collectionId"]');
+      await frame.type('input[data-test="addCollection-collectionId"]', collectionId);
+
+      // type partition key value
+      await frame.waitFor('input[data-test="addCollection-partitionKeyValue"]');
+      await frame.type('input[data-test="addCollection-partitionKeyValue"]', sharedKey);
+
+      // click submit
+      await frame.waitFor("#submitBtnAddCollection");
+      await frame.click("#submitBtnAddCollection");
+
+      // validate created
+      // open database menu
+      await frame.waitFor(`span[title="${dbId}"]`);
+      await frame.waitForSelector('div[class="splashScreen"] > div[class="title"]', { visible: true });
+
+      await frame.click(`div[data-test="${dbId}"]`);
+      await frame.waitFor(`span[title="${collectionId}"]`, { visible: true });
+      await frame.waitFor(RENDER_DELAY);
+      await frame.waitFor(`span[title="${collectionId}"]`, { visible: true });
```

Comment:
nit: consider breaking this down into two functions (one that creates and one that deletes) if you think they could be reused later in other tests.